daemon/src/battery/tplink.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{battery::BatteryState, error::RayhunterError};
+
+pub async fn get_battery_state() -> Result<BatteryState, RayhunterError> {
+    let uci_battery = tokio::process::Command::new("uci")
+        .arg("get")
+        .arg("battery.battery_mgr.power_level")
+        .output()
+        .await?;
+
+    let uci_plugged_in = tokio::process::Command::new("uci")
+        .arg("get")
+        .arg("battery.battery_mgr.is_charging")
+        .output()
+        .await?;
+
+    if !uci_battery.status.success() {
+        return Err(RayhunterError::BatteryLevelParseError);
+    }
+
+    if !uci_plugged_in.status.success() {
+        return Err(RayhunterError::BatteryPluggedInStatusParseError);
+    }
+
+    let uci_battery = String::from_utf8_lossy(&uci_battery.stdout)
+        .to_string()
+        .trim_end()
+        .parse()
+        .or(Err(RayhunterError::BatteryLevelParseError));
+
+    let uci_plugged_in = match String::from_utf8_lossy(&uci_plugged_in.stdout)
+        .to_string()
+        .chars()
+        .next()
+    {
+        Some('0') => Ok(false),
+        Some('1') => Ok(true),
```

Comment:
i think it's possible to pattern-match like `match x.as_str() { "0" => ... }` as well, would be slightly shorter